src/esp/physics/bullet/BulletPhysicsManager.cpp

```diff
@@ -181,46 +182,46 @@ void BulletPhysicsManager::stepPhysics(double dt) {
 void BulletPhysicsManager::setMargin(const int physObjectID,
                                      const double margin) {
   assertIDValidity(physObjectID);
-  static_cast<BulletRigidObject*>(existingObjects_.at(physObjectID))
+  dynamic_cast<BulletRigidObject*>(existingObjects_.at(physObjectID).get())
       ->setMargin(margin);
 }
 
 void BulletPhysicsManager::setSceneFrictionCoefficient(
     const double frictionCoefficient) {
-  static_cast<BulletRigidObject*>(sceneNode_)
-      ->setFrictionCoefficient(frictionCoefficient);
+  sceneNode_->setFrictionCoefficient(frictionCoefficient);
 }
 
 void BulletPhysicsManager::setSceneRestitutionCoefficient(
     const double restitutionCoefficient) {
-  static_cast<BulletRigidObject*>(sceneNode_)
-      ->setRestitutionCoefficient(restitutionCoefficient);
+  sceneNode_->setRestitutionCoefficient(restitutionCoefficient);
```

Comment:
The `sceneNode_` name is confusing now, as this looks like these are all methods on a SceneNode, while it's a Feature instead.